src/views/DidUpgradeExplainer/DidUpgradeExplainer.tsx

```diff
@@ -10,14 +11,39 @@ import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
 import { Avatar } from '../../components/Avatar/Avatar';
 import { LinkBack } from '../../components/LinkBack/LinkBack';
 import { Stats } from '../../components/Stats/Stats';
+import { useSwrDataOrThrow } from '../../utilities/useSwrDataOrThrow/useSwrDataOrThrow';
+import { getPromoStatus } from '../../utilities/didUpgradePromo/didUpgradePromo';
 
 interface Props {
   identity: Identity;
 }
 
-export function DidUpgradeExplainer({ identity }: Props): JSX.Element {
+export function DidUpgradeExplainer({ identity }: Props): JSX.Element | null {
   const t = browser.i18n.getMessage;
 
+  const [promoChecked, setPromoChecked] = useState(false);
+
+  const handleChecked = useCallback(() => {
+    setPromoChecked(!promoChecked);
+  }, [promoChecked]);
+
+  const promoStatus = useSwrDataOrThrow('', getPromoStatus, 'getPromoStatus');
+
+  if (!promoStatus) {
+    return null; // backend data pending
```

Comment:
I think we can show everything even while this is pending? That would be a shift in the layout, but I don’t think it’s terrible. The shift is better than not seeing anything for a while.

Reply:
The user then could potentially not know about the promo, since the checkbox could take a moment to appear.

Reply:
But I could solve that by not rendering the buttons when the data is pending.

Reply:
We’re fine if the user doesn’t learn about the promo. We’d be very upset if the backend request never ends and the user never sees any upgrade button.

Reply:
Oh yeah I see your point